### ISSUE TITLE
Incorrectly determines 'non-exhaustive pattern in `\case`' on `with` expressions

### ISSUE THREAD

**Issue by tkerber** [omelkonian/agda2rust#3]
```agda
open import Agda.Builtin.Nat using (zero; suc; _+_) renaming (Nat to ℕ)

data Bool : Set where
  true : Bool
  false : Bool

f : ℕ → Bool
f zero = true
f (suc _) = false

g : ℕ → ℕ
g x with f x
... | true = 1
... | false = x
```

Outputs:

```
*** compiling definition: test.g
 type: Agda.Builtin.Nat.Nat -> Agda.Builtin.Nat.Nat
 compiling function
 tdef: λ a →
  let b = test.f a in
  case b of
    test.Bool.true → 1
    test.Bool.false → a
 ** compiling type: Agda.Builtin.Nat.Nat
 ** compiling type: Agda.Builtin.Nat.Nat
 * compiling let: let a = test.f ^1 in
case a of
  test.Bool.true → 1
  test.Bool.false → ^1
 * compiling application: test.f $ [^1]
 * compiling case expression:
case ^1 of
  test.Bool.true → 1
  test.Bool.false → ^2
 * compiling arm (constructor): test.Bool.true
 * compiling arm (constructor): test.Bool.false
src/Agda2Rust/Convert/Terms.hs:(80,54)-(83,31): Non-exhaustive patterns in \case
```

Complains about a non-exhaustive pattern in the `with` statement. Curiously, this error does *not* occur if both branches are equal, I assume that the compiler unifies the case before it reaches this point in that case.

**Comment by omelkonian** [comment]
Simultaneously fixed in 426f8a2.

**Comment by omelkonian** [comment]
(added your examples under `test/Bool00.agda`  in c06e6c9)
